querqy-core/src/main/java/querqy/rewrite/contrib/numberunit/NumberUnitQueryCreatorSolr.java

```diff
@@ -0,0 +1,124 @@
+package querqy.rewrite.contrib.numberunit;
+
+import querqy.model.BoostQuery;
+import querqy.model.Clause;
+import querqy.model.RawQuery;
+import querqy.rewrite.contrib.numberunit.model.LinearFunction;
+import querqy.rewrite.contrib.numberunit.model.NumberUnitDefinition;
+import querqy.rewrite.contrib.numberunit.model.PerUnitNumberUnitDefinition;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+public class NumberUnitQueryCreatorSolr extends NumberUnitQueryCreator {
+
```

Comment:
It will be a bit of a pain but we should abstract away the search-engine specifics and know nothing about Solr/Lucene in querqy-core. The SearchEngineRequestAdapter might provide a kind of a `RawQueryBuilder` implementation that has methods for creating exact match query strings, OR queries etc. 

I'm open to adding a new query type, maybe 'PreprocessedQuerqy', so instead of RawQuery strings we could create Lucene queries or Elasticsearch QueryBuilder objects - but all hidden behind the SearchEngineRequestAdapter.

A last thought, we could create yet another query object model, with the SearchEngineRequestAdapter providing a search-engine specific implementation of it and then ask that model to materialise itself into a query of the search engine (cf. Query Object Pattern https://www.martinfowler.com/eaaCatalog/queryObject.html ) - but maybe that would be a bit of an overkill.

Reply:
This was the reason why I created the abstract class as an enhanced interface. The ES adaption will more or less be done by replacing this class (at least on the rewriter side). I am not sure whether the query logic can be fully covered by such a builder in this certain case, but I totally agree to your idea.